incubator/hnc/pkg/reconcilers/hnc_config_test.go

```diff
@@ -43,20 +44,25 @@ var _ = Describe("HNCConfiguration", func() {
 	ctx := context.Background()
 
 	var (
-		fooName string
-		barName string
+		fooName        string
+		barName        string
+		createdObjects []*unstructured.Unstructured
 	)
 
 	BeforeEach(func() {
 		fooName = createNS(ctx, "foo")
 		barName = createNS(ctx, "bar")
+		createdObjects = []*unstructured.Unstructured{}
 	})
 
 	AfterEach(func() {
 		// Change current config back to the default value.
 		Eventually(func() error {
 			return resetHNCConfigToDefault(ctx)
 		}).Should(Succeed())
+		for _, obj := range createdObjects {
+			k8sClient.Delete(ctx, obj)
```

Comment:
I considered using an Expect/Success here, but for the tests that test deleting an object or propagating a deletion, the test writer suddenly needs to stop using the pattern of appending the created objects which is confusing to read. I decided to leave it as a best effort cleanup here.

Reply:
A dangling object could cause other tests to fail so I wonder if we can make it a bit more robust. It's fairly easy to tell if the returned error is NotFound ([example](https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/hnc/pkg/reconcilers/hierarchical_namespace.go#L66)) so we could treat that as a success, and retry anything else.

We could do a manual retry as well, eg:
* Go through the entire list, deleting everything; make a new list only of the objects we couldn't delete
* If the new list is empty, done. Otherwise, short sleep and try again (up to two more times)